pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/HelixBrokerStarter.java

```diff
@@ -235,7 +234,7 @@ public void start()
     FunctionRegistry.init();
     _brokerRequestHandler =
         new SingleConnectionBrokerRequestHandler(_brokerConf, _routingManager, _accessControlFactory, queryQuotaManager,
-            _brokerMetrics, _propertyStore);
+            _brokerMetrics, new TableCache(_propertyStore));
```

Comment:
Just want to call out every deployment will now read the table config for the entire cluster, per broker. I think we do that for controller, and recently added for server.

Reply:
Don't we pick up all table configs to get query quotas (at least, the tables managed by the current broker).
In any case, we should restrict it to tables managed by this broker, and not all tables on the cluster.
With shared brokers in clusters that still does not help, so we need to find a better solution.

Reply:
@ianvkoeppe like we spoke:

Controller API that updates time boundary is probably the best solution. Offline push jobs must pay attention to table configs so that they can do other things that are needed for the table (partitioning, inverted index generation, etc.). It is possible for the push job to invoke the controller API to push the time boundary. This will also help in use cases that push multiple segments so that the time boundary is advanced after all segments are pushed. 

In addition to this, the brokers must also realize on their own that the time boundary should be moved forward for these tables. This is because a broker can get restarted after the push, and go into old behavior.  The broker can do this on the basis of the table config when it recognzes it has to serve a table. We alreeady read the table config to get the quota information in the broker.

We should not be changing the query. Instead, update the time boundary.